From f63a4418408179ded3f8e9362f4278cb82c574dc Mon Sep 17 00:00:00 2001
From: Rahul T R <r-ravikumar@ti.com>
Date: Fri, 26 May 2023 18:18:40 +0530
Subject: [PATCH 210/434] drm: bridge: cdns-mhdp8546: Fix bridge attach for
 no-hpd case

Currently bridge attach failed when display not connected.
Right thing to do is to mark the connector as not connected
and return success to bridge attach. Fix this by returning
0 in bridge attach

Fixes: 17638e6757bd ("drm: bridge: cdns-mhdp8546: Add support for no-hpd")
Signed-off-by: Rahul T R <r-ravikumar@ti.com>
Signed-off-by: Jayesh Choudhary <j-choudhary@ti.com>
---
 drivers/gpu/drm/bridge/cadence/cdns-mhdp8546-core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/bridge/cadence/cdns-mhdp8546-core.c b/drivers/gpu/drm/bridge/cadence/cdns-mhdp8546-core.c
index c71c402671b5..f1f0446f2146 100644
--- a/drivers/gpu/drm/bridge/cadence/cdns-mhdp8546-core.c
+++ b/drivers/gpu/drm/bridge/cadence/cdns-mhdp8546-core.c
@@ -1751,8 +1751,8 @@ static int cdns_mhdp_attach(struct drm_bridge *bridge,
 			return -ETIMEDOUT;
 		}
 
-		ret = cdns_mhdp_update_link_status(mhdp);
-		return ret;
+		cdns_mhdp_update_link_status(mhdp);
+		return 0;
 	}
 	/* Enable SW event interrupts */
 	if (hw_ready)
-- 
2.41.0

